### ISSUE TITLE
mAP is changing but not improving? 

### ISSUE THREAD

**Issue by scstu** [fizyr/keras-retinanet#985]
I have started training Retina Net on my database. I used the default configuration.  the first epoch started with these results: 
`mAP: 0.9210  loss: 1.4196 - regression_loss: 1.0643 - classification_loss: 0.3554`
after few epochs the loss decrease but the mAP is changing but in the interval between (90 - 92) but not improving. 
now "epoch 33 " results are: 
`mAP: 0.9039  loss: 0.4424 - regression_loss: 0.372 - classification_loss: 0.0672`

I am training on a dataset with approximately  22k images, ~34k instance with only class .. my validation set contain around 7k instance.. 
my hardware specifications : GPU: 1xTesla K80 with 16G ram..

what I am doing wrong? and how I can get better mAP?

**Comment by widedh** [comment]
Hi, Sorry I don't have an answer for your question. But instead I have a question, I am trying to make a GPU implementation. So I modified tha train file:
    parser.add_argument('--multi-gpu',        help='Number of GPUs to use for parallel processing.', type=int, default=1)
and I did 
python3 train.py ..... --multi-gpu-force --gpu GPU-ID
But when nvidia-smi      
+-----------------------------------------------------------------------------+
| NVIDIA-SMI 418.43       Driver Version: 418.43       CUDA Version: 10.1     |
|-------------------------------+----------------------+----------------------+
| GPU  Name        Persistence-M| Bus-Id        Disp.A | Volatile Uncorr. ECC |
| Fan  Temp  Perf  Pwr:Usage/Cap|         Memory-Usage | GPU-Util  Compute M. |
|===============================+======================+======================|
|   0  GeForce RTX 208...  Off  | 00000000:01:00.0  On |                  N/A |
| 18%   29C    P8     4W / 250W |    427MiB / 10988MiB |      **1%**      Default |
+-------------------------------+----------------------+----------------------+
                                                                               
+-----------------------------------------------------------------------------+
| Processes:                                                       GPU Memory |
|  GPU       PID   Type   Process name                             Usage      |
|=============================================================================|
|    0      1353      G   /usr/lib/xorg/Xorg                            18MiB |
|    0      1390      G   /usr/bin/gnome-shell                          57MiB |
|    0      1587      G   /usr/lib/xorg/Xorg                           147MiB |
|    0      1720      G   /usr/bin/gnome-shell                         150MiB |
|    0      4191      G   ...uest-channel-token=18265781320550942790    51MiB |
+-----------------------------------------------------------------------------+

As I see **1%** means the algorithm isn't using the GPU.. 

Could you help, please?

**Comment by hgaiser** [comment]
Isn't an mAP of 0.90 already very good? Regardless, the loss value and mAP are not directly related like you expect it to be. I can devise an output of the network that has horrible loss values, but 1.00 mAP. The other way is more difficult. If you have perfect loss value, then you also have 1.00 mAP.

Closing this as there is no issue.
